File: get player detail.py
import requests

def get_player_details(player_id):
    """
    Fetches player details including position and number of tickets of each type.
    
    :param player_id: ID of the player
    :return: Dictionary with player details
    """
    url = f"https://example.com/players/{player_id}"  # Replace with actual API endpoint
    response = requests.get(url)

    if response.status_code == 200:
        return response.json()
    else:
        return {"error": f"Failed to retrieve player details. Status Code: {response.status_code}"}

# Example usage
player_id = 201
player_details = get_player_details(player_id)
print(player_details)



import requests
from dataclasses import dataclass
from typing import Optional

@dataclass
class PlayerDetail:
    playerId: int
    playerName: str
    role: str
    colour: str
    location: int
    tickets_black: int
    tickets_yellow: int
    tickets_red: int
    tickets_green: int
    tickets_blue: int
    tickets_2x: int

def get_player_details(player_id: int) -> Optional[PlayerDetail]:
    """
    Fetches player details from the server, then converts the JSON response to a PlayerDetail object.
    
    :param player_id: ID of the player to retrieve
    :return: A PlayerDetail object, or None if request failed
    """
    url = f"https://example.com/players/{player_id}"  # Replace with actual API endpoint
    response = requests.get(url)

    if response.status_code == 200:
        data = response.json()
        # Convert the JSON dictionary to a PlayerDetail object
        player_detail = PlayerDetail(
            playerId=data.get("playerId"),
            playerName=data.get("playerName"),
            role=data.get("role"),
            colour=data.get("colour"),
            location=data.get("location"),
            tickets_black=data.get("black", 0),
            tickets_yellow=data.get("yellow", 0),
            tickets_red=data.get("red", 0),
            tickets_green=data.get("green", 0),
            tickets_blue=data.get("blue", 0),
            tickets_2x=data.get("2x", 0)
        )
        return player_detail
    else:
        print(f"Failed to retrieve player details. Status Code: {response.status_code}")
        return None

# Example usage:
if __name__ == "__main__":
    player_id = 201
    details = get_player_details(player_id)
    if details:
        print(details)

File: Backend/get player detail.js
class PlayerDetail {
    constructor(playerId, playerName, role, colour, location, tickets_black, tickets_yellow, tickets_red, tickets_green, tickets_blue, tickets_2x) {
        this.playerId = playerId;
        this.playerName = playerName;
        this.role = role;
        this.colour = colour;
        this.location = location;
        this.tickets_black = tickets_black;
        this.tickets_yellow = tickets_yellow;
        this.tickets_red = tickets_red;
        this.tickets_green = tickets_green;
        this.tickets_blue = tickets_blue;
        this.tickets_2x = tickets_2x;
    }
}

async function getPlayerDetails(playerId) {
    const url = `https://example.com/players/${playerId}`; // Replace with actual API endpoint
    
    try {
        const response = await fetch(url);
        
        if (!response.ok) {
            throw new Error(`Failed to retrieve player details. Status Code: ${response.status}`);
        }
        
        const data = await response.json();
        
        return new PlayerDetail(
            data.playerId,
            data.playerName,
            data.role,
            data.colour,
            data.location,
            data.black || 0,
            data.yellow || 0,
            data.red || 0,
            data.green || 0,
            data.blue || 0,
            data["2x"] || 0
        );
    } catch (error) {
        console.error(error.message);
        return null;
    }
}

// Example usage
(async () => {
    const playerId = 201;
    const playerDetails = await getPlayerDetails(playerId);
    if (playerDetails) {
        console.log(playerDetails);
    }
})();
